### DISCUSSION TITLE
Aria role naming mismatch on footnotes section and children <li>

### DISCUSSION THREAD

**Discussion by FionaMacNeill** [yuin/goldmark#239] [Q&A]
I have been running the deque [Axe accessibility analysis dev tools](https://www.deque.com/axe/devtools/) on my Hugo site, which uses goldmark parser by means of footnotes.  Something which has come up is the WAI Aria role for the footnotes. It seems that there is a mismatch between the container role name and it's children. To provide an example - here is the section of code from my site.

    <section class="footnotes" role="doc-endnotes">
    <hr>
    <ol>
    <li id="fn:1" role="doc-endnote">
    <p><em>Perioperative</em> ~ Encompasses the period prior to surgery, during surgery and through to the conclusion 
    of recovery at home; as defined by the Royal College of Anaesthetists (RCoA, 2014, pp. 4-5).
    <a href="#fnref:1" class="footnote-backref" role="doc-backlink">↩︎</a></p>
    </li>
    </ol>
    </section>

Based on testing with Firefox dev tools this can be resolved by removing the 's' from the section class role, which should be 'role="doc-endnote' as parent and child need to match. I also think it might be improved further if the role were applied to the containing `<ol>` rather than the section, as that is a more direct parent to child relationship (thinking about screen reader and keyboard navigation).

This is classed by deque as a critical accessibility issue, [as they outline on this advisory page](https://www.dropbox.com/s/6xexuv09689ught/Screenshot%202021-08-08%20at%2010.43.39.png?dl=0) so it would be great to fix it.

I initially posted this at the repo for the Hugo template that I use and they let me know that the issue was related to the parser. So I thought that I would ask here to check if this sounds like it could be a 'thing'.

<img width="500" alt="Screenshot 2021-08-08 at 10 43 39" src="https://user-images.githubusercontent.com/1122578/128627983-16688807-586a-4253-a2ec-560ac4ecf7f8.png">

Thanks in advance for your help/thoughts.

**Comment by yuin** [comment]
Footnotes in goldmark follows the  [Markdown Extra](https://michelf.ca/projects/php-markdown/extra/#fn-output) specification.
Markdown Extra defines output html as you mentioned.
